core/app/models/spree/store_credit.rb

```diff
@@ -150,7 +150,12 @@ def can_void?(payment)
   end
 
   def generate_authorization_code
-    "#{id}-SC-#{Time.current.utc.strftime('%Y%m%d%H%M%S%6N')}"
+    [
+      id,
+      'SC',
+      Time.current.utc.strftime('%Y%m%d%H%M%S%6N'),
+      SecureRandom.uuid
```

Comment:
Is the code displayed anywhere? Adding the UUID at the end will make it extreeeemely long. Should we be concerned about that?

Reply:
I don't think it's displayed anywhere. It's used internally when creating store credit events.